src/lib.rs

```diff
@@ -931,19 +936,18 @@ impl Config {
         let vcpkg_target = try!(find_vcpkg_target(&self, &msvc_target));
         let mut required_port_order = Vec::new();
 
+        let ports = try!(load_ports(&vcpkg_target));
+        let version = match ports.get(&port_name.to_owned()) {
+            Some(port) => port.version.clone(),
+            None => return Err(Error::LibNotFound(format!(
+                "package {} is not installed for vcpkg triplet {}",
+                port_name.to_owned(),
+                vcpkg_target.target_triplet.triplet,
+            ))),
+        };
```

Comment:
I've been pondering why this block needs moving to before the required_libs.is_empty() test, or how this relates to the same test further down around ~987.

Just considering the original code, is it intended that having a pre-set self.required_libs implies the ports shouldn't be probed at all? In that case, if this is an "override", perhaps we can't return a reliable version number here as the libs, while coming from vcpkg, are just whatever libs happened to be dumped in the aggregate install dir at vcpkg_target.lib_path, but it doesn't matter because we're still querying a port?

Either way I don't really understand the use case; I think it is left over from the old public lib_name() interface, which is documented as "should not be necessary to use" anymore and I suspect is probably now just left over spaghetti from a time when this package didn't have the capabilities to find all the libs on its own.

Like, looking at this, that test further down of self.required_libs.is_empty() is also always going to be true, isn't it?

Reply:
I believe that the block is being moved here to get the `version` variable available outside of the `is_empty()` block, since it's now required to save in the `Library` output struct.